### PR TITLE
Microsoft.Commerce: Ratecard updates

### PR DESCRIPTION
**Pull request by pargall**
- Moved Meter Region and Tags to MeterInfo
- Updated Offer Terms to acurately repersent the response

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ x] The title of the PR is clear and informative.
- [ x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x ] If applicable, the PR references the bug/issue that it fixes.
- [x ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ x] My spec meets the review criteria:
  - [x ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x ] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [x ] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@pargall__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (pearga). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by sarangan12** [comment]
@pargall 

1. Line 160: Under UsageSample, the format of the SubscriptionId is marked as uuid. Is it really a uuid? Could you please provide justification for the usage of uuid here?

2. In the consumes and produces section, you have marked both application/json and text/json. Why? Wouldn't application/json be sufficient?

**Comment by sarangan12** [comment]
@pargall 

Definitions MonetaryCredit-MonetaryCommitment-RecurringCharge, they are not used anywhere. Why are they defined?

**Comment by pargall** [comment]
Regarding the MonetaryCredit, MonetaryCommitment and, RecurringCharge. These can all be included in the OfferTerms of the ResourceRateCardInfo.  Let me know if there is a better way to represent that in swagger.

File: arm-commerce/2015-06-01-preview/swagger/commerce.json [hunk 1]
```diff
@@ -361,41 +361,48 @@
...
+          "description": "The amount of credit provided under the terms of the given offer level.",
           "type": "number",
           "format": "decimal"
+        },
+        "ExcludedMeterIds": {
+          "description": "An array of meter ids that are excluded from the given offer terms.",
+          "type": "array",
+          "items": {
+            "type": "string",
+            "format": "uuid"
```
Comment (pargall) [inline arm-commerce/2015-06-01-preview/swagger/commerce.json]: This is a list of Meter ids for which the monetary credit can not be applied against, these will be correlated with the MeterId in the MeterInfo object which is also a uuid.

**Comment by pargall** [comment]
1. Re Line 160,  That section is owned by a separate team, so I'm not sure about that section. As far as I know all Azure Subscriptions ids are guids.

2. Regarding, The produces and consumes.  I am not sure the history behind that section, presumably application/json would be sufficient. Since the api supports both MIME types is there any issue leaving it as is?

File: arm-commerce/2015-06-01-preview/swagger/commerce.json [hunk 2]
```diff
@@ -114,7 +114,7 @@
           {
             "name": "$filter",
             "in": "query",
-            "required": false,
+            "required": true,
```
Comment (amarzavery) [inline arm-commerce/2015-06-01-preview/swagger/commerce.json]: Is the api-version "2015-06-01-preview" still the same? When are you guys planning to move out of preview?

File: arm-commerce/2015-06-01-preview/swagger/commerce.json [hunk 3]
```diff
@@ -361,41 +361,48 @@
...
         "Name": {
-          "description": "Name of the offer term. For example: 'Monetary Credit', 'Monetary Commitment",
-          "type": "string"
+          "description": "Name of the offer term",
+          "type": "string",
+          "enum": [
+            "Recurring Charge",
+            "Monetary Commitment",
+            "Monetary Credit"
+          ]
```
Comment (amarzavery) [inline arm-commerce/2015-06-01-preview/swagger/commerce.json]: Please use the extension ["x-ms-enum"](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-extensions.md#x-ms-enum).

**Review by amarzavery — CHANGES_REQUESTED** [review]

**Comment by azurecla** [comment]
Hi __@pargall__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (pearga). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by amarzavery** [comment]
Hi @pargall - Can you please provide examples for request and response using the extension x-ms-examples. This will help us validate the examples against the swagger spec and it will also form the basis for customer facing REST API documentation on docs.microsoft.com website.

You can more info about the extension and an example swagger spec over [here](https://github.com/Azure/adx-documentation-pr/blob/master/tools/tools.md#swagger-model-validation).

File: arm-commerce/2015-06-01-preview/swagger/commerce.json (hunk 2 above)
Comment (pargall) [inline arm-commerce/2015-06-01-preview/swagger/commerce.json]: Yes, this is describing that version of the api.  We do have a new version out,  but I wanted to correct the existing spec before I created the new one.

There is still some churn around the UsageAggregates API that needs to be worked out prior to leaving preview.

**Comment by pargall** [comment]
@amarzavery I added the basic case to the x-ms-examples

**Review by amarzavery — APPROVED** [review]

**Pull request merged**